src/pages/index.js

```diff
@@ -12,56 +11,204 @@ import {
+  createCard,
+  fillProfileForm,
+  renderLoading,
+  checkResponse,
+} from "../utils/utils.js";
 import { UserInfo } from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
 
 import { PopupWithForm } from "../components/PopupWithForm.js";
 import { PopupWithImage } from "../components/PopupWithImage.js";
 
-const cardsSection = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardsSection.addItem(cardElement);
-    },
-  },
-  cardList
-);
-cardsSection.renderItems();
+import { Api } from "../components/Api.js";
+import { PopupWithConfirmation } from "../components/PopupWithConfirmation.js";
 //
-export const cardPopup = new PopupWithImage(modalCard);
-cardPopup.setEventListeners();
+export const previewPopup = new PopupWithImage(
+  modalCard,
+  cardModalImage,
+  cardModalTitle
+);
+previewPopup.setEventListeners();
+
+const profileInfo = new UserInfo(profileTitle, profileSubtitle, propic);
+
+const profileFormValidator = new FormValidator(config, profileEditForm);
+profileFormValidator.enableValidation();
 
-const profileEddited = new UserInfo(profileTitle, profileSubtitle);
+const cardsFormValidator = new FormValidator(config, cardsEditForm);
+cardsFormValidator.enableValidation();
+
+const propicFormValidator = new FormValidator(config, propicForm);
+propicFormValidator.enableValidation();
 
 const profilePopup = new PopupWithForm(profileEditModal, {
   handleFormSubmit: (obj) => {
-    profileEddited.setUserInfo(obj);
+    api
+      .editProfile(obj)
+      .then((r) => {
+        profilePopup.close();
+        profileInfo.setUserInfo(r);
+      })
+      .finally(() => {
+        renderLoading(false, profileEditForm, "Save");
+      });
+  },
+  handleClearError: () => {
+    profileFormValidator.clearError();
+  },
+  renderLoading: () => {
+    renderLoading(true, profileEditForm);
   },
 });
 profilePopup.setEventListeners();
 profileButtonEdit.addEventListener("click", () => {
-  fillProfileForm(profileEddited.getUserInfo());
+  fillProfileForm(profileInfo.getUserInfo());
   profilePopup.open();
 });
 
-const cardsPopup = new PopupWithForm(cardsModal, {
-  handleFormSubmit: (obj) => {
-    const cardElement = createCard(obj);
-    cardsSection.addItem(cardElement);
+//
+
+export const api = new Api(
+  {
+    baseUrl: "https://around-api.en.tripleten-services.com/v1",
+    headers: {
+      authorization: "f01264e8-2101-4ab7-b120-b09b55e63681",
+      "Content-Type": "application/json",
+    },
+  },
+  checkResponse
+);
+
+/*check:
+const newCardsSection = new Section(
+  {
+    data: "",
+    renderer: (item) => {
+      const cardElement = createCard(item);
+      cardList.append(cardElement);
+    },
+  },
+  cardList
+);
+*/
+api
+  .returnUserInfoAndCards()
+  .then((res) => {
+    profileInfo.setUserInfo(res[0]);
+
+    const newCardsSection = new Section(
```

Comment:
Only one instance of class `Section` should be created in the project because there is only **one** section for the cards.